src/data/set/finite.lean

```diff
@@ -50,6 +50,9 @@ from exists_congr (λ _, h.mem_to_finset)
 @[simp] lemma finite_empty_to_finset (h : finite (∅ : set α)) : h.to_finset = ∅ :=
 by rw [← finset.coe_inj, h.coe_to_finset, finset.coe_empty]
 
+@[simp] lemma finite.card_to_finset {s : set α} (h : finite s) :
+  h.to_finset.card = finset.card (@set.to_finset _ s h.fintype) := rfl
+
```

Comment:
This lemma seems to violate "@urkud's rule", and make it more likely that unpleasant `fintype` mismatches get introduced.

Reply:
With the old lemma, `rw ←finite.card_to_finset` would always work, but with this lemma it only works if the fintype instance is exactly this one.

Reply:
Then what's the best way to make this line work?
https://github.com/leanprover-community/mathlib/compare/pechersky/perm-support-refactor#diff-03d9da3e11d46f877e7b0e6d22ac1e2e6b800a19dddce54e5aafca535066d1dcR449

Reply:
OK, found a solution. Closing this PR.